core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -85,10 +86,13 @@ public void stopAndRemoveContainer(String containerId, String imageName) {
     }
 
     private void stopContainer(String containerId, String imageName) {
-        boolean running;
+        boolean present;
         try {
-            InspectContainerResponse containerInfo = dockerClient.inspectContainerCmd(containerId).exec();
-            running = containerInfo.getState().getRunning();
+            Optional<Container> container = dockerClient.listContainersCmd().exec()
```

Comment:
The thing I'm trying to avoid is docker-java dumping loud error logs if it tries to remove a container that doesn't exist. The name filter also seems to be responsible for this other error message which we see a lot:
```
20:35:52.348 ERROR com.github.dockerjava.core.async.ResultCallbackTemplate - Error during callback
com.github.dockerjava.api.exception.InternalServerErrorException: {"message":"invalid character '%' looking for beginning of value"}
```

I can't remember the exact condition that leads to it (even though it was only a week ago!) but found that the name filters can't be relied upon to work cleanly.